components/widgets/menu/menu_items/submenu_item.tsx

```diff
@@ -47,6 +47,8 @@ export type Props = {
     direction?: 'left' | 'right';
     openUp?: boolean;
     styleSelectableItem?: boolean;
+    renderLeft?: boolean;
+    renderRight?: boolean;
```

Comment:
I see that the `direction` prop was originally intended for controlling menu position. I would prefer to keep it that way to avoid confusion between the original prop and the two new props.

Why not just hard-code the arrow on the right regardless of what `direction`'s value is and use the `direction` prop just to control the placement of the menu? @matthewbirtch has already confirmed that the arrow should just stay on the right, so I think this would be appropriate.

Looks like `direction` is also controlling whether the keyboard shortcut uses 'left' or 'right' arrows to open the menu. I would update that to only use the right arrow shortcut regardless of direction as the arrow will now always be pointing right as well. Probably best to get @matthewbirtch to weigh in on that though.

The ideal for the menu placement would of course be to automatically detect if there is space to render the menu right and if not render it left, but that's not in scope at the moment. 🙂

Reply:
> @matthewbirtch has already confirmed that the arrow should just stay on the right.

This is correct

> Looks like direction is also controlling whether the keyboard shortcut uses 'left' or 'right' arrows to open the menu. I would update that to only use the right arrow shortcut regardless of direction as the arrow will now always be pointing right as well. Probably best to get @matthewbirtch to weigh in on that though.

Or shouldn't the submenu open when you focus on the menu item (without the need to use `Left` or `Right` arrow keys? Since this menu appears on hover with the mouse interaction, I would expect it would appear on focus with the keyboard interaction. Thoughts?

> The ideal for the menu placement would of course be to automatically detect if there is space to render the menu right and if not render it left, but that's not in scope at the moment.

100%. The intent is to display submenus on the right when there is available space to do so

Reply:
> Or shouldn't the submenu open when you focus on the menu item

Quite possibly, yes. That wouldn't be an insignificant lift though and I don't think we have time to implement and properly QA with this round of the GH. The menu system is going to be rebuilt at some point anyway, you okay to consider this later with that effort? Or maybe create a ticket to deal with after this initial GH push?

Reply:
> The menu system is going to be rebuilt at some point anyway, you okay to consider this later with that effort? Or maybe create a ticket to deal with after this initial GH push?

Yep, I'm totally fine with that.

Reply:
👍